src/go/rdctl/pkg/config/config.go

```diff
@@ -60,9 +60,20 @@ var (
 // DefineGlobalFlags sets up the global flags, available for all sub-commands
 func DefineGlobalFlags(rootCmd *cobra.Command) {
 	var err error
-	if runtime.GOOS == "linux" && isWSLDistro() {
-		if configDir, err = wslifyConfigDir(); err != nil {
-			log.Fatalf("Can't get WSL config-dir: %v", err)
+	if runtime.GOOS == "linux" {
+		if isWSLDistro() {
+			if configDir, err = wslifyConfigDir(); err != nil {
+				log.Fatalf("Can't get WSL config-dir: %v", err)
+			}
+		} else {
+			configDir = os.Getenv("XDG_DATA_HOME")
+			if configDir == "" {
+				configDir = os.Getenv("HOME")
+				if configDir == "" {
+					log.Fatalf("HOME isn't defined on this linux system.")
+				}
+				configDir = filepath.Join(configDir, ".local", "share")
+			}
```

Comment:
Not sure why `configDir` is a global variable; it seems be only used inside this function.

But I'm also confused that it is called `configDir` and not `dataDir` or `appDataDir` to make it clearer that it is no longer pointing to `os.UserConfigDir()`.

But that opens up the question if `--config-path` would have to change too, which I don't think is necessary.

Reply:
Question 1:
The docstring for `DefineGlobalFlags`:
```
// DefineGlobalFlags sets up the global flags, available for all sub-commands
```

`configDir` points to the location of `rd-engine.json`. Users can override it with the `--config-path` option.

I think some of the confusion in this whole appHome/DataHome/config area is from the definition of `os.UserConfigDir()` in the Go world. On Linux it points to `${XDG_CONFIG_HOME:-$HOME/.config}` but on macOS it points to `$HOME/Library/Application Support`